tools/svsm/Launch_SVSM.sh

```diff
@@ -0,0 +1,596 @@
+    generate_guest_ssh_keypair
+    cloud_init_create_data
+
+    
+    # For the cloud-init image, just resize the image
+    qemu-img resize "$IMAGE" "${GUEST_SIZE_GB}G"
+
+    # Add seed image option to qemu cmdline
+    add_qemu_cmdline_opts "-device scsi-hd,drive=disk1"
+    add_qemu_cmdline_opts "-drive if=none,id=disk1,format=raw,file=${LAUNCH_WORKING_DIR}/${GUEST_NAME}-seed.img"
+  fi
+
+  local guest_kernel_installed_file="${LAUNCH_WORKING_DIR}/guest_kernel_already_installed"
+   
+  if [ ! -f "${guest_kernel_installed_file}" ]; then
+    # Launch qemu cmdline
+    "${QEMU_CMDLINE_FILE}"
+
+    # Install the guest kernel, retrieve the initrd and then reboot
+    local guest_kernel_version=$(strings "${SETUP_WORKING_DIR}"/guest/linux/vmlinux | grep -a "Linux version" | head -n 1 | awk '{print $3}')
+    local guest_kernel_deb=$(echo "$(realpath "${SETUP_WORKING_DIR}"/guest/linux-image*snp-guest*.deb)" | grep -v dbg)
+    local guest_initrd_basename="initrd.img-${guest_kernel_version}"
+    local guest_vmlinuz_basename="vmlinuz-${guest_kernel_version}"
+    wait_and_retry_command "scp_guest_command ${guest_kernel_deb} ${GUEST_USER}@localhost:/home/${GUEST_USER}"
+    ssh_guest_command "sudo dpkg -i /home/${GUEST_USER}/$(basename "${guest_kernel_deb}")"
+    scp_guest_command "${GUEST_USER}@localhost:/boot/${guest_initrd_basename}" "${LAUNCH_WORKING_DIR}"
+    scp_guest_command "${GUEST_USER}@localhost:/boot/${guest_vmlinuz_basename}" "${LAUNCH_WORKING_DIR}"
+    ssh_guest_command "sudo shutdown now" || true
+    echo "true" > "${guest_kernel_installed_file}"
+
+    # Update the initrd file path and name in the guest launch source-bins file
+    sed -i -e "s|^\(INITRD_BIN=\).*$|\1\"${LAUNCH_WORKING_DIR}/${guest_initrd_basename}\"|g" "${LAUNCH_WORKING_DIR}/source-bins"    
+
+    # A few seconds for shutdown to complete
+    sleep 10    
+    return 0
+  fi
+  
+}
+
+# Function to launch QEMU VM with Coconut SVSM
+launch_qemu_svsm() {
+    echo "Launching final QEMU VM with Coconut SVSM..."
+    sudo "$QEMU_BIN" \
+      --enable-kvm \
+      -cpu EPYC-v4 \
+      -machine q35,confidential-guest-support=sev0,memory-backend=mem0 \
+      -smp 4 \
+      -m 2048 \
+      -no-reboot \
+      -vga std \
+      -nographic \
+      -object memory-backend-memfd,size=2G,id=mem0,share=true,prealloc=false,reserve=off \
+      -object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,init-flags=5,igvm-file="$IGVM_FILE" \
+      -netdev user,hostfwd=tcp::10022-:22,id=vmnic  \
+      -netdev user,id=vmnic2 -device e1000,netdev=vmnic,romfile= \
+      -device virtio-scsi-pci,id=scsi0,disable-legacy=on,iommu_platform=true \
+      -device scsi-hd,drive=disk0 \
+      -drive if=none,id=disk0,format=qcow2,file="$IMAGE" \
+      -kernel "$KERNEL_BIN" \
+      -initrd "$INITRD_BIN" \
+      -append "$GUEST_KERNEL_APPEND" \
+      -serial stdio \
+      --trace "kvm_sev*" \
+      -D /home/amd/snp/launch/qemu-trace.log \
+      -monitor none 
+      
+
+}
```

Comment:
If this is is how you're going to launch the svsm guest, what't the point of keeping the build_base_qemu_cmdline() further up?

Reply:
I can rename the function launch_qemu_svsm to launch_snp_svsm_guest(). The purpose of this function is to launch an SNP-enabled guest with SVSM-supported command options in QEMU. Additionally, I will rename setup_and_launch_svsm_guest() to launch_guest(), as this function does not require SNP to be enabled.
This is where i am using build_base_qemu_cmdline() to build basic qemu cmd. Any further comments.